pythonFiles/pytest-vscode-integration/pytest_vscode_integration.py

```diff
@@ -0,0 +1,189 @@
+    session_node["children"] = list(session_children_dict.values())
+    return session_node, errors
+
+
+def build_nested_folders(
+    file_module, file_node, created_files_folders_dict, session
+) -> TestNode:
+    prev_folder_node: TestNode = file_node
+    # Begin the i_path iteration one level above the current file.
+    iterator_path = file_module.path.parent
+    while iterator_path != session.path:
+        curr_folder_name = iterator_path.name
+        curr_folder_node = created_files_folders_dict.setdefault(
+            curr_folder_name, create_folder_node(curr_folder_name, iterator_path)
+        )
+        if prev_folder_node not in curr_folder_node["children"]:
+            curr_folder_node["children"].append(prev_folder_node)
+        iterator_path = iterator_path.parent
+        prev_folder_node = curr_folder_node
+    return prev_folder_node
+
+
+def create_test_node(test_case) -> TestItem:
+    return {
+        "name": test_case.name,
+        "path": str(test_case.path),
+        "lineno": test_case.location[1] + 1,
+        "type_": TestNodeTypeEnum.test,
+        "id_": test_case.nodeid,  # remove cast
+        "runID": test_case.nodeid,
+    }
+
+
+def create_session_node(session) -> TestNode:
+    return {
+        "name": session.name,
+        "path": str(session.path),
+        "type_": TestNodeTypeEnum.folder,
+        "children": [],
+        "id_": str(session.path),
+    }
+
+
+def create_class_node(class_module) -> TestNode:
+    return {
+        "name": class_module.name,
+        "path": str(class_module.path),
+        "type_": TestNodeTypeEnum.class_,
+        "children": [],
+        "id_": class_module.nodeid,
+    }
+
+
+def create_file_node(file_module) -> TestNode:
+    return {
+        "name": str(file_module.path.name),
+        "path": str(file_module.path),
+        "type_": TestNodeTypeEnum.file,
+        "id_": str(file_module.path),
+        "children": [],
+    }
+
+
+def create_folder_node(folderName, path_iterator) -> TestNode:
+    return {
+        "name": folderName,
+        "path": str(path_iterator),
+        "type_": TestNodeTypeEnum.folder,
+        "id_": str(path_iterator),
+        "children": [],
+    }
+
+
+class PayloadDict(TypedDict):
+    cwd: str
+    status: Literal["success", "error"]
+    tests: NotRequired[TestNode]
+    errors: NotRequired[List[str]]
+
+
+def sendPost(cwd, tests):
+    payload: PayloadDict = {"cwd": cwd, "status": "success", "tests": tests}
+    testPort = os.getenv("TEST_PORT", 45454)
+    testuuid = os.getenv("TEST_UUID")
+    addr = ("localhost", int(testPort))
+    print("sending post", addr, cwd)
+    with socket_manager.SocketManager(addr) as s:
+        data = json.dumps(payload)
+        request = f"""POST / HTTP/1.1
+Host: localhost:{testPort}
+Content-Length: {len(data)}
+Content-Type: application/json
+Request-uuid: {testuuid}
+
+{data}"""
+        result = s.socket.sendall(request.encode("utf-8"))  # type: ignore
```

Comment:
That socket only needs to be open for the send, so you should keep it open for a shorter amount of time. You are also assigning to a variable and then never using it.

```suggestion
    
    data = json.dumps(payload)
    request = f"""POST / HTTP/1.1
Host: localhost:{testPort}
Content-Length: {len(data)}
Content-Type: application/json
Request-uuid: {testuuid}

{data}"""
     with socket_manager.SocketManager(addr) as s:
        s.socket.sendall(request.encode("utf-8"))  # type: ignore
```

Reply:
And why the `type: ignore` opt-out?